src/Bicep.LangServer/Settings/ConfigurationSettingsHandler.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.JsonRpc;
+
+namespace Bicep.LanguageServer.Settings
+{
+    [Method("workspace/didChangeConfiguration", Direction.ClientToServer)]
+    public record DidChangeConfigurationObjectParams : IRequest
+    {
+        [JsonProperty("settings")]
+        public JToken? Settings;
+    }
+
+    /// <summary>
+    /// Handles settings change notification from client.
+    /// </summary>
+    public class ConfigurationSettingsHandler : IJsonRpcNotificationHandler<DidChangeConfigurationObjectParams>
+    {
+        private readonly ISettingsProvider settingsProvider;
+
+        public ConfigurationSettingsHandler(ISettingsProvider settingsProvider)
+        {
+            this.settingsProvider = settingsProvider;
+        }
+
+        Task<Unit> IRequestHandler<DidChangeConfigurationObjectParams, Unit>.Handle(DidChangeConfigurationObjectParams request, CancellationToken cancellationToken)
```

Comment:
>Handle [](http://example.com/codeflow?start=77&length=6)

Btw, do we get notified about current setting values on startup or do we only get told when config changes?

Reply:
Yes, we do.

![image](https://user-images.githubusercontent.com/30270536/222486070-ed798107-245a-48c3-b232-e3d58c95e849.png)

Reply:
Nice!